openapi.yaml

```diff
@@ -139,54 +139,72 @@ info:
     
     | Name                             | Description                                                  | Example |
     | -------------------------------- | ------------------------------------------------------------ | ------- |
-    | Access-Control-Allow-Origin      | Allowed origin for the request, including protocol, host and port. It is RECOMMENDED to return the value of the request's origin header. If no `Origin` is sent to the back-end CORS headers SHOULD NOT be sent at all. | `http://client.isp.com:80` |
-    | Access-Control-Allow-Credentials | If authorization is implemented by the back-end the value MUST be `true`. | `true` |
-    | Access-Control-Expose-Headers    | Some endpoints send non-safelisted HTTP response headers such as `OpenEO-Identifier` and `OpenEO-Costs`. All headers except `Cache-Control`, `Content-Language`, `Content-Type`, `Expires`, `Last-Modified` and `Pragma` must be listed in this header. Currently, the openEO API requires at least the following headers to be listed: `Location, OpenEO-Identifier, OpenEO-Costs, Link`. | `Location, OpenEO-Identifier, OpenEO-Costs, Link` |
+    | Access-Control-Allow-Headers     | Comma-separated list of HTTP headers allowed to be send. MUST contain at least `Authorization` if any kind of authorization is implemented by the back-end. | `Authorization, Content-Type` |
+    | Access-Control-Allow-Methods     | Comma-separated list of HTTP methods allowed to be requested. Back-ends MUST list all implemented HTTP methods for the endpoint. | `OPTIONS, GET, POST, PATCH, PUT, DELETE` |
+    | Content-Type                     | SHOULD return the content type delivered by the request that the permission is requested for. | `application/json` |
     
+    ### Example request and response
     
-    **Tip**: Most server can send the required headers and the responses to the OPTIONS requests globally. Otherwise you may want to use a proxy server to add the headers and OPTIONS responses.
+    Request:
+    
+    ```http
+    OPTIONS /api/v1/jobs HTTP/1.1
+    Host: openeo.cloudprovider.com
+    Origin: https://client.org:8080
+    Access-Control-Request-Method: POST 
+    Access-Control-Request-Headers: Authorization, Content-Type
+    ```
+    
+    Response:
+    
+    ```http
+    HTTP/1.1 204 No Content
+    Access-Control-Allow-Origin: *
+    Access-Control-Allow-Methods: OPTIONS, GET, POST, PATCH, PUT, DELETE
+    Access-Control-Allow-Headers: Authorization, Content-Type
+    Access-Control-Expose-Headers: Location, OpenEO-Identifier, OpenEO-Costs, Link
+    Content-Type: application/json
+    ```
```

Comment:
Should there also be a section about what browser based clients are expected to to do CORS-wise? Like explain the `Access-Control-Request-Method`, `Access-Control-Request-Headers` headers from the request examples?

Reply:
That is done automatically by the browsers, I don't actually need to that in the JS client. Thus I added "Note that the `Access-Control-Request-*` headers are automatically attached to the request by the browsers."